src/derivations/Segmentation.js

```diff
@@ -0,0 +1,278 @@
+import { DicomMetaDictionary } from "../DicomMetaDictionary.js";
+import DerivedPixels from "./DerivedPixels";
+import DerivedDataset from "./DerivedDataset";
+import { Normalizer } from "../normalizers.js";
+import { BitArray } from "../bitArray.js";
+
+export default class Segmentation extends DerivedPixels {
+    constructor(datasets, options = { includeSliceSpacing: true }) {
+        super(datasets, options);
+    }
+
+    derive() {
+        super.derive();
+
+        this.assignToDataset({
+            SOPClassUID: DicomMetaDictionary.sopClassUIDsByName.Segmentation,
+            Modality: "SEG",
+            SamplesPerPixel: "1",
+            PhotometricInterpretation: "MONOCHROME2",
+            BitsAllocated: "1",
+            BitsStored: "1",
+            HighBit: "0",
+            PixelRepresentation: "0",
+            LossyImageCompression: "00",
+            SegmentationType: "BINARY",
+            ContentLabel: "EXAMPLE"
+        });
+
+        let dimensionUID = DicomMetaDictionary.uid();
+        this.dataset.DimensionOrganizationSequence = {
+            DimensionOrganizationUID: dimensionUID
+        };
+        this.dataset.DimensionIndexSequence = [
+            {
+                DimensionOrganizationUID: dimensionUID,
+                DimensionIndexPointer: 6422539,
+                FunctionalGroupPointer: 6422538, // SegmentIdentificationSequence
+                DimensionDescriptionLabel: "ReferencedSegmentNumber"
+            },
+            {
+                DimensionOrganizationUID: dimensionUID,
+                DimensionIndexPointer: 2097202,
+                FunctionalGroupPointer: 2134291, // PlanePositionSequence
+                DimensionDescriptionLabel: "ImagePositionPatient"
+            }
+        ];
+
+        this.dataset.SegmentSequence = [];
+
+        // TODO: check logic here.
+        // If the referenced dataset itself references a series, then copy.
+        // Otherwise, reference the dataset itself.
+        // This should allow Slicer and others to get the correct original
+        // images when loading Legacy Converted Images, but it's a workaround
+        // that really doesn't belong here.
+        if (this.referencedDataset.ReferencedSeriesSequence) {
+            this.dataset.ReferencedSeriesSequence = DerivedDataset.copyDataset(
+                this.referencedDataset.ReferencedSeriesSequence
```

Comment:
Again, perhaps @pieper can answer this one, from the `TODO:` comment it looks like its for Legacy support for Slicer?

Reply:
Isn't that the normal meaning of TODO? 😄 

Basically this was a placeholder because at the time there wasn't any application that I could find which would understand the referenced images.  Now that we are sorting out round-trips to/from Slicer/OHIF we're in a better place.

Reply:
I missed the "TODO" part, sorry

Reply:
Slicer has been able to interpret referenced series since the QIN-HEADNECK collection release around RSNA 2015.

Reply:
Yes, but references to references is the issue: here we have a multiframe that was derived from a set of single frames, so the question is should the segmentation reference the multiframe or the original single frames.  This code was an attempt to allow loading the original singleframe datasets in Slicer even though the segmentation was defined on the multiframe.  The question (still) in my mind is if we should (1) export the segmentation and only reference the multiframe it was defined on; (2) treat the multiframe as "temporary" and export the segmentation with references to the singleframes (that's what it tries to do here); or (3) change the segmentation code to operate directly on the singleframes and not create the multiframe.

My initial preference was option (1), but since the multiframe should technically be a "legacy converted multiframe" and Slicer didn't load those, I went with option (2) instead.  But maybe option (3) is even better aligned with what a user might expect.  Clearly there's additional bookkeeping to be done to handle a wider variety of use cases.